tests/integration/test_trusted_relay.py

```diff
@@ -0,0 +1,306 @@
+import json
+import os
+from time import sleep
+
+import pytest
+from requests import HTTPError
+
+
+def test_trusted_relay_chain(mini_sentry, relay, relay_credentials):
+    """
+    Tests the happy path where we have a static relay that is registered in the
+    managed relay and sends the proper signature using the correct key pair.
+    """
+    project_id = 42
+    credentials = relay_credentials()
+    static_relays = {
+        credentials["id"]: {
+            "public_key": credentials["public_key"],
+            "internal": False,
+        },
+    }
+
+    config = mini_sentry.add_basic_project_config(project_id)
+    config["config"]["trustedRelaySettings"]["verifySignature"] = "enabled"
+    config["config"]["trustedRelays"] = [credentials["public_key"]]
+
+    managed_relay = relay(mini_sentry, static_relays=static_relays)
+
+    relay = relay(managed_relay, credentials=credentials)
+
+    relay.send_event(project_id, {"message": "trusted event"})
+    sleep(1)
+    relay.send_event(project_id, {"message": "trusted event"})
+
+    envelope = mini_sentry.captured_events.get(timeout=1)
+    event = envelope.get_event()
+    assert event["logentry"]["formatted"] == "trusted event"
+
+
+def test_expired_signature(mini_sentry, relay, events_consumer, outcomes_consumer):
+    """
+    Tests a direct request with a signature that is generated by the added trusted relay public key.
+    """
+    project_id = 42
+
+    config = mini_sentry.add_basic_project_config(project_id)
+    config["config"]["trustedRelaySettings"]["verifySignature"] = "enabled"
+    config["config"]["trustedRelays"] = ["EpWXCzYBxsX3jIp9P5YpVffwW3AIJJbB7BtWeaWM7Mk"]
+
+    events_consumer = events_consumer()
+    outcomes_consumer = outcomes_consumer()
```

Comment:
We should be able to run all these tests without processing, they will be much faster.